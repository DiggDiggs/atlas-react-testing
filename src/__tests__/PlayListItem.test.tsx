import React from "react";
import { render } from "@testing-library/react";
import PlayListItem from "../components/PlayListItem";

describe("PlayListItem Component", () => {
  it("should match the snapshot", () => {
    const mockProps = {
      title: "Test Title",
      artist: "Test Artist",
      length: "3:45",
      onClick: jest.fn(),
    };

    const { asFragment } = render(<PlayListItem {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
