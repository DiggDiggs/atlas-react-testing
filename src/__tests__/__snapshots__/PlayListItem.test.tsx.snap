// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Component should match the snapshot 1`] = `
<div
  class="mb-2 flex w-full cursor-pointer justify-between pr-3 font-medium transition-colors hover:bg-gray-200"
>
  <div
    class="wrapper text-left text-sm"
  >
    <div
      class="text-base"
    >
      Test Title
    </div>
    <div
      class="text-emerald-800"
    >
      Test Artist
    </div>
  </div>
  <div
    class="length flex items-center"
  >
    <div
      class="text-sm text-emerald-800"
    >
      3:45
    </div>
  </div>
</div>
`;
